src/main/kotlin/com/avail/interpreter/levelTwo/operation/L2_PHI_PSEUDO_OPERATION.kt

```diff
@@ -0,0 +1,360 @@
+	 *   A replacement [L2Instruction], whose operation may be either another
+	 *   `L2_PHI_PSEUDO_OPERATION` or an [L2_MOVE].
+	 */
+	fun withoutIndex(
+		instruction: L2Instruction,
+		inputIndex: Int): L2Instruction
+	{
+		val oldVector: L2ReadVectorOperand<RR, R> = instruction.operand(0)
+		val destinationReg: L2WriteOperand<R> = instruction.operand(1)
+		val newSources: MutableList<RR> = ArrayList(oldVector.elements())
+		newSources.removeAt(inputIndex)
+		val onlyOneRegister = HashSet(newSources).size == 1
+		return if (onlyOneRegister)
+		{
+			// Replace the phi function with a simple move.
+			L2Instruction(
+				instruction.basicBlock(),
+				moveOperation,
+				newSources[0],
+				destinationReg)
+		}
+		else L2Instruction(
+			instruction.basicBlock(),
+			this,
+			oldVector.clone(newSources),
+			destinationReg)
+	}
+
+	/**
+	 * Replace this phi by providing a lambda that alters a copy of the list of
+	 * [L2ReadOperand]s that it's passed.  The predecessor edges are expected to
+	 * correspond with the inputs.  Do not attempt to normalize the phi to a
+	 * move.
+	 *
+	 * @param instruction
+	 *   The phi instruction to augment.
+	 * @param updater
+	 *   What to do to a copied mutable [List] of read operands that starts out
+	 *   having all of the vector operand's elements.
+	 */
+	fun updateVectorOperand(
+		instruction: L2Instruction,
+		updater: Consumer<MutableList<RR>>)
+	{
+		assert(instruction.operation() === this)
+		val block = instruction.basicBlock()
+		val instructionIndex = block.instructions().indexOf(instruction)
+		val vectorOperand: L2ReadVectorOperand<RR, R> = instruction.operand(0)
+		val writeOperand: L2WriteOperand<R> = instruction.operand(1)
+		instruction.justRemoved()
+		val originalElements = vectorOperand.elements()
+		val passedCopy: MutableList<RR> = ArrayList(originalElements)
+		updater.accept(passedCopy)
+		val finalCopy: List<RR> = ArrayList(passedCopy)
+		val replacementInstruction = L2Instruction(
+			block, this, vectorOperand.clone(finalCopy), writeOperand)
+		block.instructions()[instructionIndex] = replacementInstruction
+		replacementInstruction.justInserted()
+	}
+
+	/**
+	 * Examine the instruction and answer the predecessor [L2BasicBlock]s
+	 * that supply a value from the specified register.
+	 *
+	 * @param instruction
+	 *   The phi-instruction to examine.
+	 * @param usedRegister
+	 *   The [L2Register] whose use we're trying to trace back to its definition.
+	 * @return
+	 *   A [List] of predecessor blocks that supplied the usedRegister as an
+	 *   input to this phi operation.
+	 */
+	fun predecessorBlocksForUseOf(
+		instruction: L2Instruction,
+		usedRegister: L2Register): List<L2BasicBlock>
+	{
+		assert(this == instruction.operation())
+		val sources: L2ReadVectorOperand<RR, R> = instruction.operand(0)
+		assert(sources.elements().size
+				   == instruction.basicBlock().predecessorEdgesCount())
+		val list: MutableList<L2BasicBlock> = ArrayList()
+		val i = MutableInt(0)
```

Comment:
MutableInt is a special implementation like Mutable<X>, which is generally unnecessary for capturing in Kotlin functions.  So here it can be `var i = 0`, and below it can be `i++`.